dbclient/ScimClient.py

```diff
@@ -108,6 +108,18 @@ def is_member_a_service_principal(member_json):
             return True
         return False
 
+    @staticmethod
+    def are_service_principal_names_unique(service_principals_log_path):
+        with open(service_principals_log_path, 'r') as fp:
+            service_principals = {}
+            for line in fp:
+                sp = json.loads(line)
+                name = sp['displayName']
+                if name in service_principals:
+                    return False
+                service_principals[name] = sp
```

Comment:
Just curious: why would we store as dict `name: sp` if we're only returning the bool? Why won't we just use a list with already parsed sp names instead?

Reply:
Simply because dicts have a faster lookup time than a list, so the `name in service_principals` is almost immediate. There's no real need to store anything as value, though